# BADM Project


### Dataset Info: https://www.kaggle.com/uciml/autompg-dataset

* Jeet Mav

In [1]:
#Importing Pandas to read dataset as a dataframe
import pandas as pd
import numpy as np

In [2]:
np.random.seed(32)

In [3]:
#Importing the dataset
df = pd.read_csv('auto-mpg.csv')

In [4]:
#Dataset info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    float64
 3   horsepower    398 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model year    398 non-null    int64  
 7   origin        398 non-null    int64  
 8   car name      398 non-null    object 
dtypes: float64(3), int64(4), object(2)
memory usage: 28.1+ KB


In [5]:
#Reading 10 Random samples from the Dataset
df.sample(10)

mpg  cylinders  displacement horsepower  weight  acceleration  \
372  27.0          4         151.0         90    2735          18.0   
325  44.3          4          90.0         48    2085          21.7   
144  31.0          4          76.0         52    1649          16.5   
192  22.0          6         250.0        105    3353          14.5   
264  18.1          8         302.0        139    3205          11.2   
107  18.0          6         232.0        100    2789          15.0   
149  24.0          4         120.0         97    2489          15.0   
342  30.0          4         135.0         84    2385          12.9   
382  34.0          4         108.0         70    2245          16.9   
243  21.5          3          80.0        110    2720          13.5   

     model year  origin              car name  
372          82       1       pontiac phoenix  
325          80       2  vw rabbit c (diesel)  
144          74       3         toyota corona  
192          76       1        chevrolet nova  
264          78       1           ford futura  
107          73       1           amc gremlin  
149          74       3           honda civic  
342          81       1      plymouth reliant  
382          82       3        toyota corolla  
243          77       3            mazda rx-4

### As we can see there are some "?" in Horsepower column we will be dropping those columns along with the column "car name" & "origin"

In [6]:
#Dropping the columns 'car name' and 'origin'
columns = ['car name', 'origin']
df.drop(columns, axis = 1,inplace = True )

In [7]:
#Printing random 10 samples from the dataframe
df.sample(10)

mpg  cylinders  displacement horsepower  weight  acceleration  \
300  23.9          8         260.0         90    3420          22.2   
114  26.0          4          98.0         90    2265          15.5   
21   24.0          4         107.0         90    2430          14.5   
332  29.8          4          89.0         62    1845          15.3   
178  23.0          4         120.0         88    2957          17.0   
317  34.3          4          97.0         78    2188          15.8   
186  27.0          4         101.0         83    2202          15.3   
346  32.3          4          97.0         67    2065          17.8   
98   16.0          6         250.0        100    3278          18.0   
144  31.0          4          76.0         52    1649          16.5   

     model year  
300          79  
114          73  
21           70  
332          80  
178          75  
317          80  
186          76  
346          81  
98           73  
144          74

In [8]:
# Creating a new dataframe where horsepower is not equal to "?"
df_new = df[df['horsepower'] != '?'] 

# Print the new dataframe shape
df_new.shape 


(392, 7)

### We have a total of 392 records (rows) and 7 properties (columns)

In [9]:
df_new['horsepower'] = df_new['horsepower'].astype('int64')

<ipython-input-9-d68539b1b7bc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new['horsepower'] = df_new['horsepower'].astype('int64')


In [10]:
#Checking for the type of horsepower
df_new.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 392 entries, 0 to 397
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           392 non-null    float64
 1   cylinders     392 non-null    int64  
 2   displacement  392 non-null    float64
 3   horsepower    392 non-null    int64  
 4   weight        392 non-null    int64  
 5   acceleration  392 non-null    float64
 6   model year    392 non-null    int64  
dtypes: float64(3), int64(4)
memory usage: 24.5 KB


In [11]:
#Converting the Dataframe to Excel for Additional analysis in MS Excel
df_new.to_excel (r'C:\Users\mavje\Desktop\Auto-MPG.xlsx', index = False, header=True)